crates/subspace-farmer/src/bin/subspace-farmer/commands/farm.rs

```diff
@@ -351,7 +352,22 @@ where
     )?;
     let mut networking_fut = Box::pin(networking_fut).fuse();
 
+    let bootstrap_fut = Box::pin({
+        let node = node.clone();
+
+        async move {
+            if let Err(err) = node.bootstrap().await {
+                warn!(?err, "DSN bootstrap failed.");
+            }
```

Comment:
It will not. Right now `NodeRunner::run` drives 6 futures forward with `select!{}` in a loop. What I'm suggesting is to add another loop just like that before, but without `self.command_receiver.next()` in it. Swarm events will still be handled, but new commands will not be processed until the very first bootstrapping is finished. When it is, you exit the first loop and start the second as usual.